pkg/metrics/grafanaorganization_collector.go

```diff
@@ -0,0 +1,103 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/giantswarm/observability-operator/api/v1alpha1"
+)
+
+// GrafanaOrganizationCollector collects metrics for GrafanaOrganization resources
+type GrafanaOrganizationCollector struct {
+	client client.Client
+}
+
+// NewGrafanaOrganizationCollector creates a new GrafanaOrganizationCollector
+func NewGrafanaOrganizationCollector(client client.Client) *GrafanaOrganizationCollector {
+	return &GrafanaOrganizationCollector{
+		client: client,
+	}
+}
+
+// CollectMetrics collects and updates all GrafanaOrganization metrics
+func (c *GrafanaOrganizationCollector) CollectMetrics(ctx context.Context) error {
+	var organizations v1alpha1.GrafanaOrganizationList
+	if err := c.client.List(ctx, &organizations); err != nil {
+		return fmt.Errorf("failed to list GrafanaOrganization resources: %w", err)
+	}
+
+	// Reset all gauge metrics
+	GrafanaOrganizationTotal.Reset()
+	GrafanaOrganizationDataSources.Reset()
+	GrafanaOrganizationTenants.Reset()
+	GrafanaOrganizationAge.Reset()
+	GrafanaOrganizationInfo.Reset()
+
+	statusCounts := map[string]int{
+		"active":  0,
+		"pending": 0,
+		"error":   0,
+	}
+
+	now := time.Now()
+
+	for _, org := range organizations.Items {
+		orgName := org.Name
+		orgIDStr := strconv.FormatInt(org.Status.OrgID, 10)
+		displayName := org.Spec.DisplayName
+		hasFinalizer := "false"
+		if controllerutil.ContainsFinalizer(&org, v1alpha1.GrafanaOrganizationFinalizer) {
+			hasFinalizer = "true"
+		}
+
+		// Determine status
+		status := "pending"
+		if org.Status.OrgID > 0 {
+			status = "active"
+		}
+		if !org.DeletionTimestamp.IsZero() {
+			status = "error" // Consider resources being deleted as in error state for monitoring
```

Comment:
This feels very sketchy, an org being deleted is not an error, it's just deleting IMO